src/disco/shred/fd_shred_tile.c

```diff
@@ -0,0 +1,694 @@
+                               *arp_sz, ctl, tsorig, tspub );
+            ctx->net_seq   = fd_seq_inc( ctx->net_seq, 1UL );
+            ctx->net_chunk = fd_dcache_compact_next( ctx->net_chunk, *arp_sz, ctx->net_chunk0, ctx->net_wmark );
+            /* TODO: Break? Limit it to one ARP requrest per loop? */
+          } else {
+            /* increment counter */
+          }
+        }
+        if( FD_UNLIKELY( !can_send_to_dest ) ) {
+          ctx->shred_dest[i].ip4  =         0U;
+          ctx->shred_dest[i].port = (ushort)0 ;
+          memset( ctx->shred_dest[i].mac_addr, 0, 6UL );
+        }
+
+        memcpy( ctx->shred_dest[i].pubkey.uc, in_dests[i].pubkey, 32UL );
+        ctx->shred_dest[i].stake_lamports = in_dests[i].stake_lamports;
+
+        memcpy( ctx->stake_weight[i].key.uc, in_dests[i].pubkey, 32UL );
+        ctx->stake_weight[i].stake = in_dests[i].stake_lamports;
+
+        staked_cnt += (ulong)(in_dests[i].stake_lamports>0UL);
+      }
+
+
+      ulong version_b = fd_mvcc_version_query( cluster_nodes_mvcc );
+      if( FD_LIKELY( version_a == version_b ) ) {
+        /* read completed cleanly */
+        fd_epoch_leaders_join( fd_epoch_leaders_new( fd_epoch_leaders_delete( fd_epoch_leaders_leave( ctx->lsched ) ), epoch, slot_start, slot_cnt, staked_cnt, ctx->stake_weight ) );
+        fd_shred_dest_join( fd_shred_dest_new( fd_shred_dest_delete( fd_shred_dest_leave( ctx->sdest ) ), ctx->shred_dest, dest_cnt, ctx->lsched, (fd_pubkey_t const *)(ctx->args->shred_signing_key+32UL) ) );
+        break;
+      }
+    }
+    ctx->prev_contact_version = fd_ulong_if( cluster_nodes_updated, version_a, ctx->prev_contact_version );
+  }
+}
+
+#define NET_IN_IDX 0
+#define MAX_SLOTS_PER_EPOCH 432000UL
+
+static void
+before_frag( void * _ctx,
+             ulong  in_idx,
+             ulong  seq,
+             ulong  sig,
+             int *  opt_filter ) {
+  (void)seq;
+
+  shred_tile_ctx_t * ctx = (shred_tile_ctx_t *)_ctx;
+
+  *opt_filter = (in_idx==NET_IN_IDX) & (fd_disco_netmux_sig_port( sig )!=ctx->args->shred_listen_port);
+}
+
+static void
+during_frag( void * _ctx,
+             ulong  in_idx,
+             ulong  sig,
+             ulong  chunk,
+             ulong  sz,
+             int *  opt_filter ) {
+  (void)sig;
+  (void)opt_filter;
+
+  shred_tile_ctx_t * ctx = (shred_tile_ctx_t *)_ctx;
+
+  if( FD_UNLIKELY( in_idx!=NET_IN_IDX ) ) {
+    /* We can just go ahead and shred it here, even though we may get
+       overrun.  If we do end up getting overrun, we just won't send
+       these shreds out and we'll reuse the FEC set for the next one.
+       From a higher level though, if we do get overrun, a bunch of
+       shreds will never be transmitted, and we'll end up producing a
+       block that never lands on chain. */
+    fd_fec_set_t * out = ctx->fec_sets + ctx->shredder_fec_set_idx;
+
+    uchar const * dcache_entry = fd_chunk_to_laddr_const( ctx->bank_shred_wksp, chunk );
```

Comment:
Can you validate that chunk is in bounds of the in wksp here? You can look at eg, dedup or verify to see how other tiles do this.